pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1383,13 +1383,21 @@ impl RemoteTimelineClient {
 
             let upload_result: anyhow::Result<()> = match &task.op {
                 UploadOp::UploadLayer(ref layer, ref layer_metadata) => {
-                    let path = layer.local_path();
+                    let local_path = layer.local_path();
+
+                    let remote_path = remote_layer_path(
+                        &self.tenant_shard_id.tenant_id,
+                        &self.timeline_id,
+                        layer_metadata.shard,
+                        &layer.layer_desc().filename(),
+                        layer_metadata.generation,
+                    );
+
                     upload::upload_timeline_layer(
-                        self.conf,
                         &self.storage_impl,
-                        path,
+                        local_path,
+                        &remote_path,
                         layer_metadata,
-                        self.generation,
                         &self.cancel,
                     )
```

Comment:
Also, why is this still taking the full `layer_metadata`? IMO it should be taking a projection type; currently, only `file_size` is used.

LayerMetadata is conceptually not needed for uploading a layer.

Reply:
This can indeed be simplified: ec7744ec6